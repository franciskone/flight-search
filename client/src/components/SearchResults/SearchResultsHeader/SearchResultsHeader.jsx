import React from 'react';
import PropTypes from 'prop-types';

import BpkLongArrowRight from 'bpk-component-icon/lg/long-arrow-right';
import STYLE from './SearchResultsHeader.scss';
import { styleGetter } from '../../../util';

const c = styleGetter(STYLE);

export const formatPassengers = (amount, cabinClass) => {
  const travellers = amount > 1
    ? `${amount} travellers`
    : `${amount} traveller`;

  return cabinClass
    ? `${travellers}, ${cabinClass}`
    : travellers;
};

const SearchResultsHeader = ({
  origin, destination, passengers, cabinClass,
}) => (
  <div className={c('SearchResultsHeader')}>
    <div className={c('SearchResultsHeader__top')}>
      {origin}
      <BpkLongArrowRight className={c('SearchResultsHeader__arrow')} /> { /* TODO Franciskone: make the icon a little bigger  */}
      {destination}
    </div>

    <div className={c('SearchResultsHeader__bottom')}>
      {formatPassengers(passengers, cabinClass)}
    </div>
  </div>);

SearchResultsHeader.propTypes = {
  origin: PropTypes.string.isRequired,
  destination: PropTypes.string.isRequired,
  passengers: PropTypes.number.isRequired,
  cabinClass: PropTypes.string,
};
SearchResultsHeader.defaultProps = {
  cabinClass: null,
};

export default SearchResultsHeader;

