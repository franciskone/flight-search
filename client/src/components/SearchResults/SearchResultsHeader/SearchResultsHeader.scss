@import '~bpk-mixins';

.SearchResultsHeader {
  background-color: $bpk-color-blue-500;
  color: $bpk-color-white;
  padding: $bpk-spacing-sm;

  &__top {
    display: flex;
    align-items: center;
    font-size: $bpk-font-size-root * 3.4;
    line-height: initial; // TODO Franciskone: spike if we can avoid this
    text-transform: uppercase;
  }

  &__arrow {
    fill: currentColor;
    margin: 0.5rem;
  }

  &__bottom {
    font-size: $bpk-font-size-base * 1.2; // TODO Franciskone: find a better way to get this font-sizes
    font-weight: 300;
  }
}