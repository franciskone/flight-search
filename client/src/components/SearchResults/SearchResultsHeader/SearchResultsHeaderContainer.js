import { connect } from 'react-redux';
import SearchResultsHeader from './SearchResultsHeader';
import {
  flightSearchQueryDestinationCodeSelector,
  flightSearchQueryOriginCodeSelector, flightSearchQueryPassengersCabinClassSelector,
  flightSearchQueryPassengersNumberSelector,
} from '../../../features/flightSearch';


const mapStateToProps = state => ({
  origin: flightSearchQueryOriginCodeSelector(state),
  destination: flightSearchQueryDestinationCodeSelector(state),
  passengers: flightSearchQueryPassengersNumberSelector(state),
  cabinClass: flightSearchQueryPassengersCabinClassSelector(state),
});

const SearchResultsHeaderContainer = connect(mapStateToProps)(SearchResultsHeader);

export default SearchResultsHeaderContainer;
