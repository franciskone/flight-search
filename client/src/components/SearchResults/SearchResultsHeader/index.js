import SearchResultsHeader from './SearchResultsHeader';

export default SearchResultsHeader;
