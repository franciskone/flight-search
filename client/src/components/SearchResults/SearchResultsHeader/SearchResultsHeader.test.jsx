import React from 'react';
import TestRenderer from 'react-test-renderer';

import SearchResultsHeader, { formatPassengers } from './SearchResultsHeader';

describe('SearchResultsHeader', () => {
  describe('component render', () => {
    it('should render correctly with cabin class', () => {
      const tree = TestRenderer.create(
        <SearchResultsHeader
          origin="ori"
          destination="des"
          passengers={1}
          cabinClass="cabinclass"
        />,
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
    it('should render correctly with cabin class and multiple travellers', () => {
      const tree = TestRenderer.create(
        <SearchResultsHeader
          origin="ori"
          destination="des"
          passengers={2}
          cabinClass="cabinclass"
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render correctly without cabin class', () => {
      const tree = TestRenderer.create(
        <SearchResultsHeader
          origin="ori"
          destination="des"
          passengers={1}
        />,
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('formatPassengers', () => {
    const TRAVELLER = 'traveller';
    const CABIN_CLASS = 'test';

    it('single passengers no cabin class', () => {
      expect(formatPassengers(1)).toBe(`1 ${TRAVELLER}`);
    });
    it('multiple passengers no cabin class', () => {
      expect(formatPassengers(2)).toBe(`2 ${TRAVELLER}s`);
    });
    it('single passengers with cabin class', () => {
      expect(formatPassengers(1, CABIN_CLASS)).toBe(`1 ${TRAVELLER}, ${CABIN_CLASS}`);
    });
    it('multiple passengers with cabin class', () => {
      expect(formatPassengers(2, CABIN_CLASS)).toBe(`2 ${TRAVELLER}s, ${CABIN_CLASS}`);
    });
  });
});

