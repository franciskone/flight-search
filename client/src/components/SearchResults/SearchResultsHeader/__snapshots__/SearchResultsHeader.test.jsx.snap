// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsHeader component render should render correctly with cabin class 1`] = `
<div
  className="SearchResultsHeader"
>
  <div
    className="SearchResultsHeader__top"
  >
    ori
    <svg
      className="SearchResultsHeader__arrow"
      height="24"
      style={
        Object {
          "height": "1.5rem",
          "width": "1.5rem",
        }
      }
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
      />
    </svg>
    des
  </div>
  <div
    className="SearchResultsHeader__bottom"
  >
    1 traveller, cabinclass
  </div>
</div>
`;

exports[`SearchResultsHeader component render should render correctly with cabin class and multiple travellers 1`] = `
<div
  className="SearchResultsHeader"
>
  <div
    className="SearchResultsHeader__top"
  >
    ori
    <svg
      className="SearchResultsHeader__arrow"
      height="24"
      style={
        Object {
          "height": "1.5rem",
          "width": "1.5rem",
        }
      }
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
      />
    </svg>
    des
  </div>
  <div
    className="SearchResultsHeader__bottom"
  >
    2 travellers, cabinclass
  </div>
</div>
`;

exports[`SearchResultsHeader component render should render correctly without cabin class 1`] = `
<div
  className="SearchResultsHeader"
>
  <div
    className="SearchResultsHeader__top"
  >
    ori
    <svg
      className="SearchResultsHeader__arrow"
      height="24"
      style={
        Object {
          "height": "1.5rem",
          "width": "1.5rem",
        }
      }
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7s0-.4-.1-.6c-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
      />
    </svg>
    des
  </div>
  <div
    className="SearchResultsHeader__bottom"
  >
    1 traveller
  </div>
</div>
`;
